apps/payments/next/app/[offeringId]/error/page.tsx

```diff
@@ -70,96 +31,91 @@ export default async function Error({ params }: { params: CheckoutParams }) {
-  const cartService = await app.getCartService();
-
-  // Note: commented out as this portion is not part of main tasks
-  // taken from T&P to assist in translations
-  // comments retained from T&P
-  // TODO - Temporary
-  // Identify an approach to ensure we don't have to perform this logic
-  // in every component/page that requires localization.
-  // const languages = headers()
-  //   .get('Accept-Language')
-  //   ?.split(',')
-  //   .map((language) => language.split(';')[0]);
-
-  // TODO
-  // Move to instantiation on start up. Ideally getBundle's, generateBundle, is only called once at startup,
-  // and then that instance is used for all requests.
-  // Approach 1 (Experimental): https://nextjs.org/docs/app/building-your-application/optimizing/instrumentation
-  // Approach 2 (Node global): https://github.com/vercel/next.js/blob/canary/examples/with-knex/knex/index.js#L13
-  // const l10n = await getBundle(languages);
+  // const cartService = await app.getCartService();
+
+  const languages = headers()
+    .get('Accept-Language')
+    ?.split(',')
+    .map((language) => language.split(';')[0]);
+  const l10n = await getBundle(languages);
+
+  const getErrorReason = (reason: string) => {
+    switch (reason) {
+      case 'iap_upgrade_contact_support':
+        return {
+          buttonFtl: 'payment-error-manage-subscription-button',
+          buttonLabel: 'Manage my subscription',
+          message:
+            'You can still get this product — please contact support so we can help you.',
+          messageFtl: 'iap-upgrade-contact-support',
+        };
+      default:
+        return {
+          buttonFtl: 'payment-error-retry-button',
+          buttonLabel: 'Try again',
+          message: 'Something went wrong. Please try again later.',
+          messageFtl: 'basic-error-message',
+        };
+    }
+  };
 
   return (
     <>
-      {/* error page header and subheader */}
-      <header className="subscription-title page-title-container desktop:pt-10">
-        <h1 className="font-semibold leading-8 mb-1 text-grey-600 text-xl">
-          Error confirming subscription…
+      <header className="page-title-container">
+        <h1 className="page-header">
+          {l10n.getMessage('subscription-error-title')?.value?.toString() ||
+            'Error confirming subscription…'}
         </h1>
-        <div className="flex items-center justify-center gap-2 text-green-900 mb-4">
-          <CheckIcon />
-          <h2 className="font-semibold text-sm">30-day money-back guarantee</h2>
+        <div className="page-subheader">
+          <Image src={checkLogo} alt="" />
+          <span className="page-subheader-text">
+            {l10n.getMessage('sub-guarantee')?.value?.toString() ||
+              '30-day money-back guarantee'}
+          </span>
         </div>
       </header>
 
-      {/* body */}
-      <article className="component-card border-t-0 mb-6 pt-4 rounded-t-lg text-grey-600 px-4 pb-12">
```

Comment:
I'd like to revisit using `<article>`. I removed it in this pull request in order to place `payment-panel` where it typically is located, which caused styling issues.

I thought we had considered `Purchase Details` as a part of the main content?

Reply:
Works for me! I would love to sit down and get a final decision on this. We halted the TW epic before we could apply all appropriate landmarks and labels.

I have mixed feelings about `payment-panel` as it is both additional and could be deemed required information. I haven't gotten a concrete answer on what is best. In mobile we can hide this information. Since the hackathon was strictly for desktop, I adjusted the page to make it the most accessible within those constraints only. Please feel free to re-add this section in.